### PR TITLE
feat(integrations): email alert link to slack

### PR DESCRIPTION
**Pull request by scefali**
This PR adds a link to slack in the issue alert emails and digests if the org does not have any alert integrations (or plugins) installed for that org/project.

Single issue:
Screen Shot 2021-05-14 at 9 53 17 AM

Digest:
Screen Shot 2021-05-14 at 9 55 52 AM

The `Setup in Slack` button brings you to the integration detail page for slack with the `alert_email` referrer.

### CODE REVIEW

**Comment by MeredithAnya** [comment]
@scefali this seems like it could be really annoying for people who intentionally never want to set up slack, like they are on free or they use a different alerting tool

**Comment by scefali** [comment]
@MeredithAnya We'll, the main idea is to get free orgs to upgrade. But I'll bring your comment to the folks behind the idea and see what they say.

**Comment by MeredithAnya** [comment]
> @MeredithAnya We'll, the main idea is to get free orgs to upgrade. But I'll bring your comment to the folks behind the idea and see what they say.

well then maybe gate this to organizations on free plans

**Comment by ajjindal** [comment]
@scefali I agree with the part that we can start by limiting this to free orgs and then expand in future.

**Comment by scefali** [comment]
@MeredithAnya @ajjindal Limiting it to just free orgs adds complexity because we have to get subscriptions involved and doesn't make it less annoying IMO. What if we checked if the org has any alerting integrations installed instead?

**Comment by ajjindal** [comment]
As long as we can check that the orgs don't have any notification (Teams, Slack, PagerDuty etc), we should be fine on duplicating.

**Comment by scefali** [comment]
@MeredithAnya Updated the PR

File: src/sentry/notifications/utils/__init__.py [hunk 1]
```diff
@@ -200,9 +222,31 @@ def has_integrations(organization: Organization, project: Project) -> bool:
 
     project_plugins = plugins.for_project(project, version=1)
     organization_integrations = Integration.objects.filter(organizations=organization).first()
+    # TODO: fix because project_plugins is an iterator and thus always truthy
     return bool(project_plugins or organization_integrations)
 
 
+def is_alert_rule_integration(provider: IntegrationProvider) -> bool:
+    return any(feature == IntegrationFeatures.ALERT_RULE for feature in provider.features)
+    # return any(lambda x: x == IntegrationFeatures.ALERT_RULE, provider.features)
```
Comment (MeredithAnya) [inline src/sentry/notifications/utils/__init__.py:231]: was this just left over?

File: src/sentry/notifications/utils/__init__.py [hunk 2]
```diff
@@ -200,9 +222,31 @@ def has_integrations(organization: Organization, project: Project) -> bool:
...
+    providers = filter(is_alert_rule_integration, list(integrations.all()))
+    provider_keys = map(lambda x: cast(str, x.key), providers)
+    if Integration.objects.filter(organizations=org, provider__in=provider_keys).exists():
+        return True
+
+    # check plugins
+    from sentry.plugins.base import plugins
+
+    project_plugins = plugins.for_project(project, version=None)
+    return any(plugin.get_plugin_type() == "notification" for plugin in project_plugins)
```
Comment (MeredithAnya) [inline src/sentry/notifications/utils/__init__.py:247]: are webhooks counted in this?

Comment (scefali) [reply src/sentry/notifications/utils/__init__.py:247]: @MeredithAnya Yup, it's a subclass of `NotificationPlugin`

File: src/sentry/notifications/utils/__init__.py (hunk 1 above)
Comment (scefali) [reply src/sentry/notifications/utils/__init__.py]: @MeredithAnya fixed

**Review by MeredithAnya — APPROVED** [review]
I think the code looks good, but I think "Setup" should maybe be "Set up" since we are using it as a verb and not like "its a setup" . Approving though cause that's a trivial change

**Pull request merged**